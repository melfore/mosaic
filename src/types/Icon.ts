import { LoadableType } from "./Base";

export enum Icons {
  account = "account",
  add = "add",
  apps = "apps",
  block = "block",
  brightness = "brightness",
  business = "business",
  chart_timeline = "chart_timeline",
  check = "check",
  checkbox = "checkbox",
  checkbox_empty = "checkbox_empty",
  clock = "clock",
  close = "close",
  delete = "delete",
  down = "down",
  edit = "edit",
  error = "error",
  filter = "filter",
  first = "first",
  home = "home",
  last = "last",
  left = "left",
  list = "list",
  list_check = "list_check",
  mail = "mail",
  mail_open = "mail_open",
  menu = "menu",
  notifications = "notifications",
  open_new = "open_new",
  pause = "pause",
  play = "play",
  print = "print",
  refresh = "refresh",
  right = "right",
  save = "save",
  search = "search",
  send = "send",
  settings = "settings",
  up = "up",
  up_arrow = "up_arrow",
  warn = "warn",
}

export enum IconSize {
  default = "default",
  large = "large",
  small = "small",
}

interface IconForwardedType {
  className?: string;
  color?: string;
  ref?: any;
}

export interface IconType extends LoadableType {
  forwarded?: IconForwardedType;
  name: Icons;
  size?: IconSize;
}
