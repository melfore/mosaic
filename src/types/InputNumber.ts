import { InputType } from "./Input";

export interface InputNumberType extends InputType {
  integer?: boolean;
  minValue?: number;
  maxValue?: number;
  onChange?: (value: number | null) => void | any;
  value?: number | null;
}
