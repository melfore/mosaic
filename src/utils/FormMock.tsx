import React, { FC, ReactElement, useState } from "react";

interface FormMockType {
  onInputChange: Function;
  inputValue: null | number | string;
  onChangePropName?: string;
  valuePropName?: string;
}

const FormMock: FC<FormMockType> = ({
  children,
  onInputChange,
  inputValue,
  onChangePropName = "onChange",
  valuePropName = "value",
}) => {
  const [value, setValue] = useState(inputValue);
  const wrappedOnChange = (value: null | number | string) => {
    setValue(value);
    onInputChange(value);
  };

  return React.cloneElement(children as ReactElement<any>, {
    [onChangePropName]: wrappedOnChange,
    [valuePropName]: value,
  });
};

export default FormMock;
