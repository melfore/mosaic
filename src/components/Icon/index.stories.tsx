import React from 'react';
import {  } from '@storybook/addon-actions';
import { select } from '@storybook/addon-knobs';
import { Icons, IconSize } from '../../types/Icon';
import { DOCS_PAGE_STRUCTURE, StoriesWrapper } from '../../utils/storybook';
import Icon from '.';

export default {
  title: 'Icon',
  component: Icon,
  parameters: {
    ...DOCS_PAGE_STRUCTURE,
  },
}

export const Canvas = () => (
  <Icon
    dataCy="icon"
    name={select('icon', Icons, Icons.add)}
    size={select('size', IconSize, IconSize.default)}
  />
);

export const Size = () => (
  <StoriesWrapper>
    <Icon
      dataCy="icon-send"
      name={Icons.send}
      size={IconSize.small}
    />
    <Icon
      dataCy="icon-send"
      name={Icons.send}
    />
    <Icon
      dataCy="icon-send"
      name={Icons.send}
      size={IconSize.large}
    />
  </StoriesWrapper>
);

const allIcons = Object
  .keys(Icons)
  .map((icon: string) => (
    <div className="icon-wrapper">
      <Icon
        dataCy={`icon-${icon}`}
        name={Icons[icon]}
      />
      <span>
        {icon}
      </span>
    </div>
  ));

export const AllIcons = () => (
  <StoriesWrapper>
    {allIcons}
  </StoriesWrapper>
);
