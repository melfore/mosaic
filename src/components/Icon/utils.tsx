import React from "react";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUISendIcon from "@material-ui/icons/Send";
import { Icons } from "../../types/Icon";

export const iconsCatalog = {
  [Icons.add]: <MUIAddIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.playlistAddCheck]: <MUIPlaylistAddCheckIcon />,
  [Icons.send]: <MUISendIcon />,
};
