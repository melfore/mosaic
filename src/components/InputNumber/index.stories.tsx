import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Icons } from "../../types/Icon";
import { formDecorator } from "../../utils/mocks/FormMock";
import { localeDecorator, MessageMock } from "../../utils/mocks/LocaleMock";
import getDocsPage from "../../utils/stories";

import InputNumber, { DATA_CY_DEFAULT, DATA_CY_SHORTCUT, InputNumberWithProps, LOCALIZABLE_PROPS } from ".";

const COMPONENT_NAME = "InputNumber";
InputNumber.displayName = COMPONENT_NAME;
InputNumberWithProps.displayName = COMPONENT_NAME;

export default {
  title: "Inputs/InputNumber",
  component: InputNumberWithProps,
  decorators: [formDecorator, localeDecorator],
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Text Field",
          url: "https://mui.com/components/text-fields/",
        },
        component: COMPONENT_NAME,
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          dataCyShortcut: DATA_CY_SHORTCUT,
        },
        localizableProps: LOCALIZABLE_PROPS,
      }),
    },
  },
} as ComponentMeta<typeof InputNumberWithProps>;

const Template: ComponentStory<typeof InputNumberWithProps> = (args) => (
  <InputNumber {...args} dataCy={DATA_CY_DEFAULT} />
);

export const Primary = Template.bind({});
Primary.args = {
  label: "Label",
  value: 5,
};

export const Adornment = Template.bind({});
Adornment.args = {
  ...Primary.args,
  adornment: {
    icon: Icons.close,
  },
};

export const AdornmentClickable = Template.bind({});
AdornmentClickable.args = {
  ...Primary.args,
  adornment: {
    icon: Icons.close,
    onClick: () => {},
  },
};

export const Decimal = Template.bind({});
Decimal.args = {
  ...Primary.args,
  integer: false,
  value: 5.275,
};

export const Disabled = Template.bind({});
Disabled.args = {
  ...Primary.args,
  disabled: true,
};

export const Localized = Template.bind({});
Localized.args = {
  ...Primary.args,
  localized: true,
  label: MessageMock.inputNumber,
  placeholder: MessageMock.placeholder,
};

export const MinMax = Template.bind({});
MinMax.args = {
  ...Primary.args,
  maxValue: 1000,
  minValue: 20,
};

export const Nullable = Template.bind({});
Nullable.args = {
  ...Primary.args,
  value: null,
};

export const Placeholder = Template.bind({});
Placeholder.args = {
  label: "Label",
  placeholder: "Enter value...",
  value: undefined,
};

export const Required = Template.bind({});
Required.args = {
  ...Primary.args,
  required: true,
};

export const SizeSmall = Template.bind({});
SizeSmall.args = {
  ...Primary.args,
  size: "small",
};

export const Styled = Template.bind({});
Styled.args = {
  ...Primary.args,
  style: {
    color: "red",
    fontWeight: "bold",
    fontSize: "large",
    textAlign: "center",
  },
};

export const VariantFilled = Template.bind({});
VariantFilled.args = {
  ...Primary.args,
  variant: "filled",
};

export const VariantStandard = Template.bind({});
VariantStandard.args = {
  ...Primary.args,
  variant: "standard",
};
