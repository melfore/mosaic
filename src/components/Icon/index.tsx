import React, { FC } from "react";
import { IconType, IconSize } from "../../types/Icon";
import { iconsCatalog } from "./utils";

const Icon: FC<IconType> = ({ dataCy, forwarded = {}, name, size = IconSize.default }) => {
  const icon = iconsCatalog[name];
  return React.cloneElement(icon, { ...forwarded, "data-cy": dataCy, fontSize: size });
};

export default Icon;
