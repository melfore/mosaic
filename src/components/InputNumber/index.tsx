import React, { FC, useState, useEffect } from "react";
import { styled } from "@material-ui/core/styles";
import MUITextField from "@material-ui/core/TextField";
import { InputDataType, InputSize, InputVariant } from "../../types/Input";
import { InputNumberType } from "../../types/InputNumber";

const StyledMUITextField = styled(MUITextField)({
  width: "100%",
});

const getControlledValue = (value: number | null): any => {
  return value === null ? "" : value;
};

const getBoundedValue = (value: number | null, bounds: any): number | null => {
  const { minValue, maxValue } = bounds;
  if (value === null || (value >= minValue && value <= maxValue)) {
    return value;
  }

  return value < minValue ? minValue : maxValue;
};

const getNumericValue = (value: string, integer: boolean): number | null => {
  const numericValue = integer ? parseInt(value, 10) : parseFloat(value);
  return isNaN(numericValue) ? null : numericValue;
};

const isOutsideBounds = (value: number | null, bounds: any): boolean => {
  const { minValue, maxValue } = bounds;
  return value !== null && (value < minValue || value > maxValue);
};

/**
 * InputText component made on top of `@material-ui/core/TextField`
 */
const InputNumber: FC<InputNumberType> = ({
  dataCy,
  disabled = false,
  initialValue = null,
  integer = true,
  // TODO#lb: implement labelId
  label,
  minValue = Number.MIN_SAFE_INTEGER,
  maxValue = Number.MAX_SAFE_INTEGER,
  onChange = undefined,
  required = false,
  shrink = true,
  size = InputSize.default,
  variant = InputVariant.default,
}) => {
  const [bounds, setBounds] = useState({ minValue, maxValue });
  const [value, setValue] = useState(getBoundedValue(initialValue, bounds));
  useEffect(() => setBounds({ minValue, maxValue }), [minValue, maxValue]);
  useEffect(() => setValue(initialValue), [initialValue]);

  const onChangeHandler = (event: any) => {
    const numericValue = getNumericValue(event.target.value, integer);
    setValue(getBoundedValue(numericValue, bounds));
    if (onChange) {
      onChange(value);
    }
  };

  return (
    <StyledMUITextField
      disabled={disabled}
      error={isOutsideBounds(value, bounds)}
      InputLabelProps={{
        shrink,
      }}
      inputProps={{
        "data-cy": dataCy,
        style: { textAlign: "right" },
      }}
      InputProps={{
        readOnly: disabled,
      }}
      label={label}
      margin="normal"
      onChange={onChangeHandler}
      required={required}
      size={size}
      type={InputDataType.number}
      variant={variant}
      value={getControlledValue(value)}
    />
  );
};

export default InputNumber;
