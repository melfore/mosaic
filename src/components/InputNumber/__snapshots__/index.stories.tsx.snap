// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs/InputNumber Adornment smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r1:"
         id=":r1:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-mui-classname">
    <input aria-invalid="false"
           id=":r1:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-mui-classname">
      <button class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
              tabindex="-1"
              type="button"
              disabled
              data-cy="icon-button"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CloseIcon"
             data-cy="icon-button-icon"
        >
          <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber AdornmentClickable smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r2:"
         id=":r2:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-mui-classname">
    <input aria-invalid="false"
           id=":r2:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <div class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-mui-classname">
      <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-mui-classname"
              tabindex="0"
              type="button"
              data-cy="icon-button"
      >
        <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
             focusable="false"
             aria-hidden="true"
             viewbox="0 0 24 24"
             data-testid="CloseIcon"
             data-cy="icon-button-icon"
        >
          <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
          </path>
        </svg>
      </button>
    </div>
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Decimal play-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r3:"
         id=":r3:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r3:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="5.275"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Disabled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r4:"
         id=":r4:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl Mui-readOnly MuiInputBase-readOnly css-mui-classname">
    <input aria-invalid="false"
           disabled
           id=":r4:"
           readonly
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-readOnly MuiInputBase-readOnly css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Localized smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r5:"
         id=":r5:-label"
  >
    Campo numerico
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r5:"
           placeholder="Inserire valore..."
           type="number"
           data-cy="locale.inputNumber"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Campo numerico
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber MinMax smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r6:"
         id=":r6:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r6:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Nullable smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r7:"
         id=":r7:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r7:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Placeholder smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r8:"
         id=":r8:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r8:"
           placeholder="Enter value..."
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Primary play-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r0:"
         id=":r0:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r0:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Required smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":r9:"
         id=":r9:-label"
  >
    Label
    <span aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-mui-classname"
    >
       *
    </span>
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":r9:"
           required
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label *
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber SizeSmall smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":ra:"
         id=":ra:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-mui-classname">
    <input aria-invalid="false"
           id=":ra:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber Styled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mui-classname"
         data-shrink="true"
         for=":rb:"
         id=":rb:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":rb:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiOutlinedInput-input css-mui-classname"
           value="5"
           style="text-align: center; width: 100%; color: red; font-weight: bold; font-size: large;"
    >
    <fieldset aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mui-classname"
    >
      <legend class="css-mui-classname">
        <span>
          Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Inputs/InputNumber VariantFilled smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-filled css-mui-classname"
         data-shrink="true"
         for=":rc:"
         id=":rc:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":rc:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiFilledInput-input css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
  </div>
</div>
`;

exports[`Inputs/InputNumber VariantStandard smoke-test 1`] = `
<div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-mui-classname"
     style="text-align: right; width: 100%;"
>
  <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-mui-classname"
         data-shrink="true"
         for=":rd:"
         id=":rd:-label"
  >
    Label
  </label>
  <div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-mui-classname">
    <input aria-invalid="false"
           id=":rd:"
           type="number"
           data-cy="input-number"
           class="MuiInputBase-input MuiInput-input css-mui-classname"
           value="5"
           style="text-align: right; width: 100%;"
    >
  </div>
</div>
`;
